### PR TITLE
Refactor cert args

### PR DESCRIPTION
**Pull request by akallabeth**
Added new option /cert that unifies all.
    
* The mess with /cert-tofu, /cert-ignore et al is now unified in
      a single option.
* Added the option to add fingerprint:<hash>:<hex string> multiple
      times to /cert to build a list of accepted certificate
      fingerprints
* Added a deprecation warning to older /cert-* options
* [IMPORTANT] changes hash used in fingerprint file to `sha256` as `sha1` is broken. This will invalidate all stored certificates.

@jol64 I think this option is what you were looking for?

### CODE REVIEW

**Comment by jol64** [comment]
looks very promissing. Will download next nightly (tomorrow) and test it.
Thanks, Joachim

**Comment by jol64** [comment]
> Refer to this link for build results (access rights to CI server needed):
> https://ci.freerdp.com//job/PullRequestTester/4578/

windows builds are aborted..

**Comment by jol64** [comment]
all of the builds referenced are aborted at least for windows.

**Comment by akallabeth** [comment]
@freerdp-bot test

File: libfreerdp/crypto/crypto.c [hunk 1]
```diff
@@ -215,17 +215,50 @@ void crypto_reverse(BYTE* data, int length)
...
-	BYTE fp[EVP_MAX_MD_SIZE];
-	X509_digest(xcert, EVP_sha1(), fp, &fp_len);
-	fp_buffer = (char*)calloc(fp_len + 1, 3);
 
-	if (!fp_buffer)
+	fp = crypto_cert_hash(xcert, hash, &fp_len);
+	if (!fp)
 		return NULL;
 
+	fp_buffer = calloc(fp_len * 3 + 1, sizeof(char));
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply libfreerdp/crypto/crypto.c]: wanted to be on the safe side ;)

**Comment by jol64** [comment]
what is the exact syntax expected? I tried /cert:fingerprint:xx-yy-...  with and without -, but wfreerdp did not show a window but instead issued help information..

**Comment by akallabeth** [comment]
@jol64 `/cert:fingerprint:sha256:0123456...ABc,sha512:abcABF...123`

**Comment by jol64** [comment]
thanks for the update.

- imho the help is misleading. should probably read something like /cert:fingerprint:<algo>:<hash>[,..]
- I had to specify the hash (either sha1 or sha256) using colons to separate the bytes, very much like it is shown in the dialog (except for the dialog does not show the algorithm used)
- the dialog imho should show the algorithm in order to avoid the confusion and maybe also both sha1 and sha256 hashes, as windows uses sha1 ("thumbprint") and freerdp uses sha256.

**Comment by akallabeth** [comment]
@jol64 working on some performance and security related stuff, so this is lower priority for me, but pull requests are always welcome.

**Pull request merged**